### PR TITLE
Fix issue #1188: vmgroup is not removable when vms are added from both ESXs

### PR DESCRIPTION
**Pull request by shaominchen**
Note: This PR was created for a clean commit history. See original PR #1214 for test result and previous review comments)

### CODE REVIEW

File: esx_service/utils/test_utils.py [hunk 1]
```diff
@@ -163,3 +163,11 @@ def create_default_tenant_and_privileges(test_obj):
...
         test_obj.assertEqual(error_info, None)
 
+def cleanup_tenant(name):
+    error_info, vms = auth_api._tenant_vm_ls(name)
+    if vms:
+        vm_names = [vm_name for (_, vm_name) in vms]
+        auth_api._tenant_vm_rm(name=name,
+                               vm_list=vm_names)
+    auth_api._tenant_rm(name=name,
+                        remove_volumes=True)
```
Comment (pshahzeb) [inline esx_service/utils/test_utils.py]: nit: no new line at the end.

Comment (shaominchen) [inline esx_service/utils/test_utils.py]: Addressed.

**Comment by shaominchen** [comment]
@pshahzeb @shuklanirdesh82 The CI test has passed: https://ci.vmware.run/vmware/docker-volume-vsphere/95
Can you please add a ship-it?

File: esx_service/cli/vmdkops_admin_test.py [hunk 2]
```diff
@@ -753,6 +748,10 @@ def test_tenant(self):
 
 
         # tenant rm to remove the tenant
+        error_info = auth_api._tenant_vm_rm(name=self.tenant1_new_name,
+                                            vm_list=vm_list)
+        self.assertEqual(None, error_info)
+
         error_info = auth_api._tenant_rm(
```
Comment (shuklanirdesh82) [inline esx_service/cli/vmdkops_admin_test.py]: can we use `test_utils.cleanup_tenant` to keep it consistent?

Comment (shaominchen) [inline esx_service/cli/vmdkops_admin_test.py]: Sure, will address.

**Comment by shuklanirdesh82** [comment]
Thanks @shaominchen for addressing quickly!

**Pull request merged**